src/theory/theory_engine.cpp

```diff
@@ -311,34 +305,45 @@ void TheoryEngine::preRegister(TNode preprocessed) {
       Debug("theory") << "TheoryEngine::preRegister: " << preprocessed
                       << std::endl;
       Assert(!expr::hasFreeVar(preprocessed));
+
       // Pre-register the terms in the atom
-      Theory::Set theories = NodeVisitor<PreRegisterVisitor>::run(d_preRegistrationVisitor, preprocessed);
+      Theory::Set theories = NodeVisitor<PreRegisterVisitor>::run(
+          d_preRegistrationVisitor, preprocessed);
       theories = Theory::setRemove(THEORY_BOOL, theories);
-      // Remove the top theory, if any more that means multiple theories were involved
+      // Remove the top theory, if any more that means multiple theories were
+      // involved
       bool multipleTheories = Theory::setRemove(Theory::theoryOf(preprocessed), theories);
+#ifdef CVC4_ASSERTIONS
```

Comment:
I prefer to use `if (Configuration::isAssertionBuild()) { ... }` instead of using `#ifdefs` because it is more idiomatic.